src/adapter/src/catalog/apply.rs

```diff
@@ -988,6 +1022,13 @@ fn sort_updates(updates: Vec<StateUpdate>) -> Vec<StateUpdate> {
         }
     }
 
+    soft_assert_no_log!(
+        updates
+            .windows(2)
+            .all(|window| window[0].ts == window[1].ts),
```

Comment:
FWIW we have an [`all_equal(...)`](https://github.com/MaterializeInc/materialize/blob/main/src/ore/src/iter.rs#L32-L43) method in `mz_ore`